near-sdk/src/promise.rs

```diff
@@ -529,3 +542,13 @@ impl<T: borsh::BorshSerialize> borsh::BorshSerialize for PromiseOrValue<T> {
         }
     }
 }
+
+impl<T: JsonSchema> JsonSchema for PromiseOrValue<T> {
+    fn schema_name() -> String {
+        format!("PromiseOrValue{}", T::schema_name())
+    }
+
+    fn json_schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        T::json_schema(gen)
```

Comment:
Note that right now nothing restricts a developer from having the promise return a different schema type than `T`.

Reply:
Yeah I also thought about this. Returning different things feels like a strange pattern to me, but I guess since we are not discouraging this pattern I should not presume that there is only one possible return type. I could represent this as an equivalent of `Either<Any, T>` or just simply `Any`. WDYT?

Reply:
Well, it is not really fallible as it is non-restrictive to just `T`, so I don't think `Result` suits unless I am missing something here

Reply:
I think that making the assumption as one type is fine, it would nudge developers in the correct pattern, but I'm just making the comment so we don't ignore the cases where maybe someone might return different types that serialize as the same thing. Can't immediately think of an implication that would be an issue, though